### ISSUE TITLE
Let phing recognize git environment variables

### ISSUE THREAD

**Issue by marvil07** [acquia/blt#1249]
My system information:
* Operating system type: *nix
* BLT version: 8.6.15

When I run this command:
```
export GIT_AUTHOR_EMAIL=foo@example.com
blt deploy
```

blt uses the global git config user.email as the email set to the author inside the build repository.

I expected blt to use any of the git environment variables, e.g. `GIT_AUTHOR_EMAIL`, `GIT_COMMITTER_EMAIL`, and `EMAIL`; but maybe that is a by design.

For completion, git name can be overriden with `GIT_AUTHOR_NAME` and `GIT_COMMITTER_NAME`.

A work-around for now: override blt `deploy` phing target to set git email, e.g.
```
    <if>
      <isset property="deploy.gitEmail"/>
      <then>
        <exec command="git config user.email '${deploy.gitEmail}'" dir="${deploy.dir}" />
      </then>
    </if>
```

**Comment by grasmash** [comment]
Why not just `git config user.name email` prior to executing `blt deploy` ?

**Comment by grasmash** [comment]
I believe this is resolved by #1323.

You may use project.local.yml to set environment specific values.

**Comment by marvil07** [comment]
@grasmash I'm glad to see that feature getting in, but hopefully, we can split that configuration and document it (see #1340), a PR for that proposed on #1341.

Also, I would say that the capability of reading all git environment variables may still be desired, but I will let you decide if you want to reopen this ticket for that.
